Sources/AblyChat/RoomLifecycleManager.swift

```diff
@@ -11,31 +17,57 @@ internal protocol RoomLifecycleContributorChannel: Sendable {
 
     var state: ARTRealtimeChannelState { get async }
     var errorReason: ARTErrorInfo? { get async }
+
+    /// Equivalent to subscribing to a `RealtimeChannelProtocol` object’s state changes via its `on(_:)` method. The subscription should use the ``BufferingPolicy.unbounded`` buffering policy.
+    ///
+    /// It is marked as `async` purely to make it easier to write mocks for this method (i.e. to use an actor as a mock).
+    func subscribeToState() async -> Subscription<ARTChannelStateChange>
 }
 
-internal actor RoomLifecycleManager<Channel: RoomLifecycleContributorChannel> {
-    /// A realtime channel that contributes to the room lifecycle.
-    internal struct Contributor {
-        /// The room feature that this contributor corresponds to. Used only for choosing which error to throw when a contributor operation fails.
-        internal var feature: RoomFeature
+/// A realtime channel that contributes to the room lifecycle.
+internal protocol RoomLifecycleContributor: Sendable {
+    associatedtype Channel: RoomLifecycleContributorChannel
 
-        internal var channel: Channel
+    /// The room feature that this contributor corresponds to. Used only for choosing which error to throw when a contributor operation fails.
+    var feature: RoomFeature { get }
+    var channel: Channel { get }
+
+    /// Informs the contributor that there has been a break in channel continuity, which it should inform library users about.
+    ///
+    /// It is marked as `async` purely to make it easier to write mocks for this method (i.e. to use an actor as a mock).
+    func emitDiscontinuity(_ error: ARTErrorInfo) async
+}
+
+internal actor RoomLifecycleManager<Contributor: RoomLifecycleContributor> {
+    /// Stores manager state relating to a given contributor.
+    private struct ContributorAnnotation {
+        // TODO: Not clear whether there can be multiple or just one (asked in https://github.com/ably/specification/pull/200/files#r1781927850)
+        var pendingDiscontinuityEvents: [ARTErrorInfo] = []
```

Comment:
Did you see the comment on `contributorAnnotations` saying 

```
The annotation at a given index belongs to the element of ``contributors`` at the same index.
```

I could have instead combined the contributor and its annotation into some other struct (not sure of the name) and stored those structs in a single array. The reason I didn't was that I wanted to make the `contributors` array a `let` to make it clear that the contributors don't change over the lifetime of the room lifecycle manager.

Reply:
Makes sense, bu it looks a bit "scary" to me "what if indexes messed up somehow". It always better to have a strong bond between objects rather then to rely on indexes. Could be a dictionary with a key taken from a contributor in some way.

Reply:
That's a fair point. I've changed it so that the contributors have to conform to `Identifiable`, and am now storing the state in a dictionary keyed by contributor `id` instead.